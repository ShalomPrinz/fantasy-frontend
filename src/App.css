.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.overlap-above {
  position: relative;
  top: -4vh;
  box-shadow: 0 0 20px 2px black;
}

.shadow-below {
  box-shadow: 0 3px 10px 0 black;
}

.bg-default {
  background-color: #0E1E5B;
  color: white;
}

.bg-light-danger {
  background-color: #ff4e4e;
}

.centered-flex {
  align-items: center;
  display: flex;
  justify-content: center;
}
