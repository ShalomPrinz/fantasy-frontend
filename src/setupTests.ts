// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { ReactElement } from 'react';

import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

export type TestRendererJSON = null | ReactTestRendererJSON | ReactTestRendererJSON[];
export const getJSON = (Component: ReactElement) => renderer.create(Component).toJSON()
