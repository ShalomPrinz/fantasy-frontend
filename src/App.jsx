import React from "react";

import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import { NavBar } from "./components";
import { Home, UserTeam } from "./pages";

import "./App.css";

function App() {
  return (
    <>
      <NavBar />
      <ToastContainer 
        autoClose="2500" 
        pauseOnFocusLoss="false"
      />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/team" element={<UserTeam user={{name: "Shalom"}} />} />
      </Routes>
    </>
  );
}

export default App;
