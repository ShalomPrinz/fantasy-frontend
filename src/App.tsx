import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import { NavBar } from "./components";
import { Home, Login, Register, UserTeam } from "./pages";
import { Player, Team, User } from "./types";

import "./App.css";

const players: Player[] = [
  { id: 0, name: "Ter Stegen", role: "GK", team: "Barcelona" },
  { id: 1, name: "Pique", role: "DEF", team: "Barcelona" },
  { id: 2, name: "Kounde", role: "DEF", team: "Barcelona" },
  { id: 3, name: "Araujo", role: "DEF", team: "Barcelona" },
  { id: 4, name: "Bellingham", role: "MID", team: "Dortmund" },
  { id: 5, name: "Mokuku", role: "ATT", team: "Dortmund" },
];

const team = new Team(players);

function App() {
  const user = new User("Shalom", team);

  return (
    <>
      <NavBar />
      <ToastContainer autoClose={2500} pauseOnFocusLoss={false} />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/team" element={<UserTeam user={user} />} />
        <Route path="/register" element={<Register />} />
        <Route path="/login" element={<Login />} />
      </Routes>
    </>
  );
}

export default App;
