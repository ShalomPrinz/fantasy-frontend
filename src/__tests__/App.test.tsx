import { BrowserRouter } from 'react-router-dom';

import App from '../App';
import { getJSON, TestRendererJSON } from '../setupTests';

let tree: TestRendererJSON = null;

beforeEach(() => {
    tree = null
})

describe('App', () => {
    it('should render app component', () => {
        tree = getJSON(
            <BrowserRouter>
                <App />
            </BrowserRouter>
        );
        expect(tree).toMatchSnapshot();
    });
})