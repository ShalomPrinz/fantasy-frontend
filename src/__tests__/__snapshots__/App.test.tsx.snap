// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app component 1`] = `
Array [
  <nav
    className="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      className="navbar-brand fs-1 active"
      href="/"
      onClick={[Function]}
    >
      <img
        className="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars "
        color="white"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="navbar-nav"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <a
          className="nav-item nav-links fw-light"
          href="/team"
          onClick={[Function]}
        >
          My Team
        </a>
        <a
          aria-current="page"
          className="nav-item nav-links fw-light active"
          href="/"
          onClick={[Function]}
        >
          Second
        </a>
        <a
          aria-current="page"
          className="nav-item nav-links fw-light active"
          href="/"
          onClick={[Function]}
        >
          Third
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="Toastify"
  />,
  <div
    className="position-relative text-center overflow-hidden"
  >
    <img
      alt="Original Soccer Grass"
      className=""
      src="wide_grass.jpg"
    />
    <div
      className="position-absolute top-50 start-50 translate-middle text-white w-100"
    >
      <h1
        className="fw-bold"
      >
        Champions League Fantasy.
      </h1>
      <h4>
        Play Now
      </h4>
    </div>
  </div>,
  <div
    className="bg-white overlap-above rounded-top mx-5"
  >
    <div
      className="container d-flex flex-wrap justify-content-center text-white"
    >
      <div
        className="pt-3 m-4 w-25 card-bg-image card"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trophy fa-6x"
          data-icon="trophy"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 0H176c-26.5 0-48.1 21.8-47.1 48.2c.2 5.3 .4 10.6 .7 15.8H24C10.7 64 0 74.7 0 88c0 92.6 33.5 157 78.5 200.7c44.3 43.1 98.3 64.8 138.1 75.8c23.4 6.5 39.4 26 39.4 45.6c0 20.9-17 37.9-37.9 37.9H192c-17.7 0-32 14.3-32 32s14.3 32 32 32H384c17.7 0 32-14.3 32-32s-14.3-32-32-32H357.9C337 448 320 431 320 410.1c0-19.6 15.9-39.2 39.4-45.6c39.9-11 93.9-32.7 138.2-75.8C542.5 245 576 180.6 576 88c0-13.3-10.7-24-24-24H446.4c.3-5.2 .5-10.4 .7-15.8C448.1 21.8 426.5 0 400 0zM48.9 112h84.4c9.1 90.1 29.2 150.3 51.9 190.6c-24.9-11-50.8-26.5-73.2-48.3c-32-31.1-58-76-63-142.3zM464.1 254.3c-22.4 21.8-48.3 37.3-73.2 48.3c22.7-40.3 42.8-100.5 51.9-190.6h84.4c-5.1 66.3-31.1 111.2-63 142.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
             
            Create your league
             
          </div>
          <p
            className="fw-light card-text"
          >
             
            play the ultimate fantasy game with friends
             
          </p>
        </div>
      </div>
      <div
        className="pt-3 m-4 w-25 card-bg-image card"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ranking-star fa-6x"
          data-icon="ranking-star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M353.8 54.1L330.2 6.3c-3.9-8.3-16.1-8.6-20.4 0L286.2 54.1l-52.3 7.5c-9.3 1.4-13.3 12.9-6.4 19.8l38 37-9 52.1c-1.4 9.3 8.2 16.5 16.8 12.2l46.9-24.8 46.6 24.4c8.6 4.3 18.3-2.9 16.8-12.2l-9-52.1 38-36.6c6.8-6.8 2.9-18.3-6.4-19.8l-52.3-7.5zM256 256c-17.7 0-32 14.3-32 32V480c0 17.7 14.3 32 32 32H384c17.7 0 32-14.3 32-32V288c0-17.7-14.3-32-32-32H256zM32 320c-17.7 0-32 14.3-32 32V480c0 17.7 14.3 32 32 32H160c17.7 0 32-14.3 32-32V352c0-17.7-14.3-32-32-32H32zm416 96v64c0 17.7 14.3 32 32 32H608c17.7 0 32-14.3 32-32V416c0-17.7-14.3-32-32-32H480c-17.7 0-32 14.3-32 32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
             
            Podium
             
          </div>
          <p
            className="fw-light card-text"
          >
             
            observe some interesting stats
             
          </p>
        </div>
      </div>
      <div
        className="pt-3 m-4 w-25 card-bg-image card"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-image fa-6x"
          data-icon="image"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 60.7 28.7 32 64 32H448c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM323.8 202.5c-4.5-6.6-11.9-10.5-19.8-10.5s-15.4 3.9-19.8 10.5l-87 127.6L170.7 297c-4.6-5.7-11.5-9-18.7-9s-14.2 3.3-18.7 9l-64 80c-5.8 7.2-6.9 17.1-2.9 25.4s12.4 13.6 21.6 13.6h96 32H424c8.9 0 17.1-4.9 21.2-12.8s3.6-17.4-1.4-24.7l-120-176zM112 192c26.5 0 48-21.5 48-48s-21.5-48-48-48s-48 21.5-48 48s21.5 48 48 48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
             
            Another thing
             
          </div>
          <p
            className="fw-light card-text"
          >
             
            wow another one
             
          </p>
        </div>
      </div>
    </div>
    <div
      className="text-center fw-bold fs-3"
    >
      Simple. Fast. Addictive.
    </div>
  </div>,
]
`;
